browser/brave_wallet/brave_wallet_tab_helper_browsertest.cc

```diff
@@ -126,3 +167,54 @@ IN_PROC_BROWSER_TEST_F(BraveWalletTabHelperBrowserTest,
   base::RunLoop().RunUntilIdle();
   ASSERT_FALSE(tab_helper->IsShowingBubble());
 }
+
+IN_PROC_BROWSER_TEST_F(BraveWalletTabHelperBrowserTest,
+                       ClosePopupsWhenSwitchTabs) {
+  content::WebContents* active_contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(active_contents);
+  tab_helper->ShowApproveWalletBubble();
+  EXPECT_TRUE(tab_helper->IsShowingBubble());
+  auto* panel_contents = tab_helper->GetBubbleWebContentsForTesting();
+  WaitForLoadStop(panel_contents);
+  tab_helper->ClosePanelOnDeactivate(false);
+  auto popup1_id =
+      OpenPanelPopup(https_server()->GetURL("a.com", "/popup.html"),
+                     panel_contents, tab_helper);
+  auto popup2_id =
+      OpenPanelPopup(https_server()->GetURL("b.com", "/popup.html"),
+                     panel_contents, tab_helper);
+  EXPECT_TRUE(tab_helper->IsShowingBubble());
+  chrome::NewTab(browser());
+  base::RunLoop().RunUntilIdle();
+  ASSERT_FALSE(tab_helper->IsShowingBubble());
+  ASSERT_FALSE(GetWebContentsFromTabId(popup1_id, browser()->profile()));
+  ASSERT_FALSE(GetWebContentsFromTabId(popup2_id, browser()->profile()));
+  EXPECT_EQ(tab_helper->GetPopupIdsForTesting().size(), 0u);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveWalletTabHelperBrowserTest, ClosePopupsWithBubble) {
+  content::WebContents* active_contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  auto* tab_helper =
+      brave_wallet::BraveWalletTabHelper::FromWebContents(active_contents);
+  tab_helper->ShowApproveWalletBubble();
+  EXPECT_TRUE(tab_helper->IsShowingBubble());
+  auto* panel_contents = tab_helper->GetBubbleWebContentsForTesting();
+  WaitForLoadStop(panel_contents);
+  tab_helper->ClosePanelOnDeactivate(false);
+  auto popup1_id =
+      OpenPanelPopup(https_server()->GetURL("a.com", "/popup.html"),
+                     panel_contents, tab_helper);
+  auto popup2_id =
+      OpenPanelPopup(https_server()->GetURL("b.com", "/popup.html"),
+                     panel_contents, tab_helper);
+  EXPECT_TRUE(tab_helper->IsShowingBubble());
+  tab_helper->CloseBubble();
+  base::RunLoop().RunUntilIdle();
+  ASSERT_FALSE(tab_helper->IsShowingBubble());
+  ASSERT_FALSE(GetWebContentsFromTabId(popup1_id, browser()->profile()));
+  ASSERT_FALSE(GetWebContentsFromTabId(popup2_id, browser()->profile()));
```

Comment:
We also need to make sure non panel initiated popup won't be closed by accident.

Reply:
done